src/main/java/org/jenkinsci/plugins/workflow/multibranch/SuppressBranchIndexingBuildsJobProperty.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.multibranch;
+
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Queue;
+import jenkins.branch.BranchIndexingCause;
+import jenkins.model.OptionalJobProperty;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.List;
+
+public class SuppressBranchIndexingBuildsJobProperty extends OptionalJobProperty<WorkflowJob> {
+
+    @DataBoundConstructor
+    public SuppressBranchIndexingBuildsJobProperty() {
+    }
+
+    @Extension
+    @Symbol("suppressBranchIndexingBuilds")
+    public static class DescriptorImpl extends OptionalJobPropertyDescriptor {
+
+        @Override public String getDisplayName() {
+            return "Suppress builds triggered by branch indexing";
+        }
```

Comment:
Still kinda confused - how can I tell the difference between a `WorkflowJob` from a `WorkflowMultiBranchProject` or otherwise?